hack/deployer/runner/tanzu.go

```diff
@@ -442,4 +443,54 @@ func (t *TanzuDriver) restoreInstallerState() error {
 		WithoutStreaming().Run()
 }
 
+func (t *TanzuDriver) Cleanup(prefix string, olderThan time.Duration) error {
+	if err := t.loginToAzure(); err != nil {
+		return err
+	}
+	sinceDate := time.Now().Add(-olderThan)
+
+	params := map[string]interface{}{
+		"Date":                 sinceDate.Format(time.RFC3339),
+		"Location":             t.plan.Tanzu.Location,
+		"E2EClusterNamePrefix": prefix,
+	}
+
+	resourceGroups, err := azure.Cmd("group", "list",
+		"--query", fmt.Sprintf("[?location=='%s']", t.plan.Tanzu.Location),
+		"--query", fmt.Sprintf("[?contains(name,'%s-tanzu')]", prefix),
+		"| jq -r '.[].name'").OutputList()
+	if err != nil {
+		return err
+	}
+
+	for _, rg := range resourceGroups {
+		params["ResourceGroup"] = rg
+
+		clustersToDelete, err := azure.Cmd("resource", "list",
+			"-l", t.plan.Tanzu.Location,
+			"-g", rg,
+			`--resource-type "Microsoft.Compute/virtualMachines"`,
+			"--query", "[?tags.project == 'eck-ci']",
+			"| jq -r --arg d", sinceDate.Format(time.RFC3339),
+			fmt.Sprintf(`'map(select((.createdTime | . <= $d) and (.name|test("%s-tanzu"))))|.[].name'`, prefix),
+			fmt.Sprintf("| grep -o '%s-tanzu-[a-z]*-[0-9]*' | sort | uniq", prefix)).OutputList()
+		if err != nil {
+			return fmt.Errorf("while running az resource list command: %w", err)
+		}
+
+		for _, cluster := range clustersToDelete {
+			t.plan.ClusterName = cluster
+			t.plan.Tanzu.ResourceGroup = cluster
```

Comment:
```suggestion
			t.plan.Tanzu.ResourceGroup = rg
```

Reply:
I've updated this, even though resourcegroup == cluster in tanzu, this makes it more clear.

Also, I've continued on failure for all cluster types.